shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/metrics/MetricsServiceImplTest.java

```diff
@@ -19,37 +19,47 @@
 import static org.mockito.ArgumentCaptor.forClass;
 import static org.mockito.ArgumentMatchers.anyList;
 import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 @QuarkusTest
 public class MetricsServiceImplTest {
 
     @Inject
     OperatorMetricsService metricsService;
 
-    @InjectMock
+    @InjectMock(convertScopes = true)
```

Comment:
I have never used this param and I don't know the behaviour, is this a fix/improvement related with the PR?

Reply:
It forces `@Singleton` scoped beans to be converted to `@ApplicationScoped` since the former is not supported by `@InjectMock`. It is an interesting scenario: `MeterRegistry` is implemented by a producer method in `io.quarkus:quarkus-micrometer-registry-prometheus` that we have as a dependency in our `shard-operator` to support the export of Metrics for Prometheus. The producer method produces a `@Singleton` scoped bean. What I however do not understand is _why_ it _seemed_ to work OK on `main`.... That said it seemed lucky the whole test worked on `main`  as the `verify` was being called incorrectly.

Reply:
@danielezonca Interestingly I cannot get the test to run on either `main` or an old branch (before any refactoring). Weird.

Reply:
Nice :)
[Quarkus documentation](https://quarkus.io/guides/cdi#applicationscoped-and-singleton-look-very-similar-which-one-should-i-choose-for-my-quarkus-application) clearly states that `Singleton` doesn't work with `QuarkusMock` (because it is not a proxy) so I have no idea 🤔 
It might be that this test is not executed in CI for some reason?